Detailed docs for creation, removal, update (#26)
cowprotocol/cow-amm: docs/amm.md

Example set-up transactions: [step 1](0xfb0d01c5edbf1dd80c3d9f8a54e9a4057db2419ec50045e32161044bab5a06f3), [step 2 to 4](https://app.safe.global/transactions/tx?safe=eth:0xBEEf5aFE88eF73337e5070aB2855d37dBF5493A4&id=multisig_0xBEEf5aFE88eF73337e5070aB2855d37dBF5493A4_0x2bb29ac594fd562981ac1221596ed6b471a459615bb3b8c8ebd403e9fad079af), [step 5](https://app.safe.global/transactions/tx?safe=eth:0xBEEf5aFE88eF73337e5070aB2855d37dBF5493A4&id=multisig_0xBEEf5aFE88eF73337e5070aB2855d37dBF5493A4_0x1651687da34e9a7039e3a41e81e90cde74c1bfeeb0c9ec3dac0b7432dcbb4300).

### Step 1: deploy a Safe

The CoW AMM funds will be stored in a Safe, and every pair of tokens requires its dedicated safe.
All interactions of the owner with the AMM will go through the Safe interface.
There are no requirements on the ownership structure of the Safe, but keep in mind that the owners have full access to the funds.

You can create a Safe following the instruction on [the dedicated page on the Safe interface](https://app.safe.global/new-safe/create).

> You should not use this Safe for other purposes but running the CoW AMM: the token balances represent the state of the AMM and should not be altered outside of the normal AMM trading process.

Reviewer: ```suggestion
You should not use this Safe for purposes other than running a CoW AMM: the token balances represent the state of the AMM and should not be altered outside of the normal AMM trading process.
```

> The same Safe must **not** be used as the AMM for multiple symultaneous pairs.

Reviewer: ```suggestion
The same Safe must **not** be used as the AMM for multiple simultaneous pairs.
```

### Step 2: switch to the extensible fallback handler

> CoW Protocol relies on ERC-1271 signatures to validate smart-contract orders.

Reviewer: ```suggestion
CoW Protocol supports validation of smart-contract orders via ERC-1271 signatures.
```

Signature verification for `ComposableCoW` orders (like the order used in the CoW AMM) is handled by a dedicated [fallback handler](https://help.safe.global/en/articles/40838-what-is-a-fallback-handler-and-how-does-it-relate-to-safe), the `ExtensibleFallbackHandler`.

You can find more details on the design of `ComposableCoW` in the [dedicated page of the CoW Protocol documentation](https://docs.cow.fi/cow-protocol/reference/contracts/periphery/composable-cow#conditional-order-verification-flow).

> On the AMM Safe interface, create a new transaction on the transaction builder with the following parameters:

Reviewer: ```suggestion
On the AMM Safe{Wallet} interface, create a new transaction on the transaction builder with the following parameters:
```

Author: Context: we want to distinguish between Safe frontend and Safe wallet. The Safe logo looks similar to `Safe(Wallat)` but since they just use "Safe Wallet" in their [docs](https://safe.global/wallet) I'm going to go for that instead.

```suggestion
On the AMM Safe Wallet interface, create a new transaction on the transaction builder with the following parameters:
```
